mongo_connector/doc_managers/algolia_doc_manager.py

```diff
@@ -250,7 +250,8 @@ def apply_update(self, doc, update_spec):
                     doc[attr] = None
         return doc
 
-    def update(self, doc, update_spec, namespace = None, timestamp = None):
+    def update(self, document_id, update_spec, namespace = None, timestamp = None):
+        doc = self.index.getObject(str(document_id))
```

Comment:
We already use the partial update to avoid the call to get the object, we also override the apply_update method to handle the deleted attributes so we can put an empty dict instead of the original doc.

But there is one corner case, If your data is not flat, the default function "apply_update" will raise an error if you try to modify a nested attribute because it won't be able to access it.

With the getObject, the connector will be really slow and the number of api call will double for the update.